test/test.jwtaccess.ts

```diff
@@ -51,6 +60,25 @@ describe('.getRequestMetadata', () => {
     assert.strictEqual(email, payload.sub);
     assert.strictEqual(testUri, payload.aud);
   });
+
+  it('should return a cached token on the second request', () => {
+    const res2 = client.getRequestMetadata(testUri);
+    assert.strictEqual(res, res2);
+  });
+
+  it('should not return cached tokens older than an hour', () => {
+    const realDateNow = Date.now;
+    try {
+      // go forward in time one hour (plus a little)
+      Date.now = () => realDateNow() + (1000 * 60 * 60) + 10;
+      const res2 = client.getRequestMetadata(testUri);
+      assert.notEqual(res, res2);
+    } catch (e) {
+      assert.fail(e);
```

Comment:
If anything throws, doesn't the test fail automatically? I think you can just delete the `try-catch` from here.

Reply:
I did that to make sure that we return Date.now back to the original function.

Reply:
How does it work if you call `assert.fail(e)` on exceptions?

Did you mean to do
```typescript
   const realDateNow = Date.now;
   try {
     // go forward in time one hour (plus a little)
     Date.now = () => realDateNow() + (1000 * 60 * 60) + 10;
     const res2 = client.getRequestMetadata(testUri);
     assert.notEqual(res, res2);
   } finally {
     // return date.now to it's normally scheduled programming
     Date.now = realDateNow;
   }
```
?

Reply:
Fair point :)